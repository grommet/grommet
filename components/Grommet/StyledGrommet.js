"use strict";

exports.__esModule = true;
exports.StyledGrommet = void 0;
var _styledComponents = _interopRequireWildcard(require("styled-components"));
var _utils = require("../../utils");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
var fullStyle = function fullStyle(full) {
  if (full === 'min') return (0, _styledComponents.css)(["min-height:100vh;"]);
  return (0, _styledComponents.css)(["width:100vw;height:100vh;overflow:auto;"]);
};
var StyledGrommet = exports.StyledGrommet = _styledComponents["default"].div.withConfig(_utils.styledComponentsConfig).withConfig({
  displayName: "StyledGrommet",
  componentId: "sc-19lkkz7-0"
})(["", " ", " ", " ", " ", ""], function (props) {
  return !props.plain && _utils.baseStyle;
}, function (props) {
  return props.full && fullStyle(props.full);
}, function (props) {
  return props.theme.global.font.face;
}, function (props) {
  return props.theme.grommet.extend;
}, function (props) {
  return props.cssVars && Object.keys(props.theme.global.colors).filter(function (k) {
    return typeof props.theme.global.colors[k] === 'string';
  }).map(function (k) {
    return "--" + k + ": " + props.theme.global.colors[k] + ";";
  }).join('\n');
});