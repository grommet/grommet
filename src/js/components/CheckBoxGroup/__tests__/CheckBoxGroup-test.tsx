import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { axe } from 'jest-axe';

import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';

import { Grommet } from '../../Grommet';
import { CheckBoxGroup } from '..';

describe('CheckBoxGroup', () => {
  test('should have no accessibility violations', async () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup options={['First', 'Second']} />
      </Grommet>,
    );

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  test('options renders', () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup options={['First', 'Second']} />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders without grommet wrapper', () => {
    const { container } = render(
      <CheckBoxGroup options={['First', 'Second']} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('value renders', () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup value={['First']} options={['First', 'Second']} />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('initial value renders', () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup
          value={['Wuhan', 'Jerusalem']}
          options={[
            { label: 'Maui', value: 'Maui' },
            { label: 'Jerusalem', value: 'Jerusalem' },
            { label: 'Wuhan', value: 'Wuhan' },
          ]}
        />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('disabled renders', () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup disabled options={['First', 'Second']} />
        <CheckBoxGroup options={[{ label: 'First', disabled: true }]} />
        <CheckBoxGroup options={[{ label: 'First', disabled: true }]} />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('onChange', () => {
    const onChange = jest.fn();
    const { container, getByText } = render(
      <Grommet>
        <CheckBoxGroup
          options={[
            { label: 'first-label', value: 'First' },
            { label: 'second-label', value: 'Second' },
          ]}
          onChange={onChange}
        />
      </Grommet>,
    );
    fireEvent.click(getByText('first-label'));
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('onClick for check and uncheck a CheckBox', () => {
    const onClick = jest.fn();
    const { container, getByText } = render(
      <Grommet>
        <CheckBoxGroup
          options={[
            { label: 'first-label', value: 'First' },
            { label: 'second-label', value: 'Second' },
          ]}
          onClick={onClick}
        />
      </Grommet>,
    );
    fireEvent.click(getByText('first-label'));
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.click(getByText('first-label'));
    expect(onClick).toHaveBeenCalledTimes(2);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('labelKey', () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup
          labelKey="labelKeyTest"
          options={[
            { labelKeyTest: 'first-label', value: 'First' },
            { labelKeyTest: 'second-label', value: 'Second' },
          ]}
        />
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('valueKey', () => {
    const { container } = render(
      <Grommet>
        <CheckBoxGroup
          valueKey="valueKeyTest"
          options={[
            { label: 'first-label', valueKeyTest: 'First' },
            { label: 'second-label', valueKeyTest: 'Second' },
          ]}
        />
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('custom theme', () => {
    const customTheme = {
      checkBoxGroup: {
        container: {
          gap: 'large',
          margin: {
            vertical: 'small',
          },
        },
      },
    };

    const { container } = render(
      <Grommet theme={customTheme}>
        <CheckBoxGroup
          valueKey="valueKeyTest"
          options={[
            { label: 'first-label', valueKeyTest: 'First' },
            { label: 'second-label', valueKeyTest: 'Second' },
          ]}
        />
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('no duplicated key error', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation();

    render(
      <Grommet>
        <CheckBoxGroup
          value={['yes', 'yes-again']}
          options={[
            { label: 'Yes!', value: 'yes' },
            { label: 'Yes!', value: 'yes-again' },
          ]}
        />
      </Grommet>,
    );

    expect(errorSpy).not.toHaveBeenCalledWith(
      expect.stringMatching('same key'),
      expect.stringMatching('Yes!'),
      expect.anything(),
    );

    errorSpy.mockReset();
    errorSpy.mockRestore();
  });

  test('checked warning', () => {
    const warnSpy = jest.spyOn(console, 'warn').mockImplementation();

    render(
      <Grommet>
        <CheckBoxGroup
          options={[
            { label: 'first-label', value: 'First', checked: true },
            { label: 'second-label', value: 'Second' },
          ]}
        />
      </Grommet>,
    );

    expect(warnSpy).toHaveBeenCalled();
    warnSpy.mockReset();
    warnSpy.mockRestore();
  });

  test('defaultValue renders', () => {
    const { asFragment } = render(
      <Grommet>
        <CheckBoxGroup options={['First', 'Second']} defaultValue={['First']} />
      </Grommet>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
