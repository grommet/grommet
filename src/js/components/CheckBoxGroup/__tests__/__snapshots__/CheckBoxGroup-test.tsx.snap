// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup custom theme 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 gXZjYl StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eWqFqG"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup defaultValue renders 1`] = `
<DocumentFragment>
  <div
    class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
      role="group"
    >
      <label
        class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
        >
          <input
            checked=""
            class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
            type="checkbox"
          />
          <div
            class="StyledBox-sc-13pk1d4-0 kZSzDp StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
          >
            <svg
              class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 fTWMAU"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
            >
              <path
                d="M6,11.3 L10.3,16 L18,6.2"
                fill="none"
              />
            </svg>
          </div>
        </div>
        <span>
          First
        </span>
      </label>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
      />
      <label
        class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
        >
          <input
            class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
            type="checkbox"
          />
          <div
            class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
          />
        </div>
        <span>
          Second
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CheckBoxGroup disabled renders 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 gfAmwd"
      disabled=""
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
        disabled=""
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 hPiJDG"
          disabled=""
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
          disabled=""
        />
      </div>
      <span>
        First
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 gfAmwd"
      disabled=""
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
        disabled=""
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 hPiJDG"
          disabled=""
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
          disabled=""
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 gfAmwd"
      disabled=""
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
        disabled=""
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 hPiJDG"
          disabled=""
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
          disabled=""
        />
      </div>
      <span>
        First
      </span>
    </label>
  </div>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 gfAmwd"
      disabled=""
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
        disabled=""
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 hPiJDG"
          disabled=""
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
          disabled=""
        />
      </div>
      <span>
        First
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup initial value renders 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        Maui
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          checked=""
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kZSzDp StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 fTWMAU"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        Jerusalem
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          checked=""
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kZSzDp StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 fTWMAU"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        Wuhan
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup labelKey 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onChange 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kZSzDp StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 fTWMAU"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onClick for check and uncheck a CheckBox 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 gAEdcj StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
    tabindex="0"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kZSzDp StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 fTWMAU"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onClick for check and uncheck a CheckBox 2`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 gAEdcj StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
    tabindex="0"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup options renders 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        First
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup renders without grommet wrapper 1`] = `
<div
  class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
  role="group"
>
  <label
    class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
    >
      <input
        class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
        type="checkbox"
      />
      <div
        class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
      />
    </div>
    <span>
      First
    </span>
  </label>
  <div
    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
  />
  <label
    class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
    >
      <input
        class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
        type="checkbox"
      />
      <div
        class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
      />
    </div>
    <span>
      Second
    </span>
  </label>
</div>
`;

exports[`CheckBoxGroup value renders 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          checked=""
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kZSzDp StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 fTWMAU"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        First
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup valueKey 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fHTwQM"
>
  <div
    class="StyledBox-sc-13pk1d4-0 klipST StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cvOsLl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 jaPbMk"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 llqfhu StyledCheckBox-sc-1dbk5ju-6 iybiqv"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 BGjjG"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kIGOa-d StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iXXfEH"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;
